### PR TITLE
Document the examples, including Docker instructions; add BUILDING.md, update all READMEs

### LINKED ISSUE #72: Document ways for a user to test/run the Arrow collector
The user experience for building and running an Arrow collector is currently very opaque.

The problem is exacerbated by `go.mod` problems. We are using `replace` statements in our multi-module repositories to enable development, an old solution to a problem currently addressed by `go.work` files. 

We are unsure whether checking-in a go.work file is a good plan, but it would apparently improve the situation. 

Beyond cleanups of this sort, we need a page describing how a user can quick-start from a collector build configuration, to an initial yaml file, to first data flow.

### PR DESCRIPTION
**Pull request by jmacd**
This PR (at last!) fixes a number of outstanding Golang module issues. We now use a checked-in go.work file.

The important aspect of this change, for the go.work approach to work, involves using the OTC `builder` with its `--skip-get-modules` flag set. This allows the go.work file to be used instead, and we now call `go work sync` instead. 

There are no more `replace` statements in any of the go.mod files, so single-line commands will now work (after the next release). 

The new BUILDING.md instructions explains how to build either with a local Golang toolchain, or using a Docker Golang toolchain. Each example has been tested and updated for a number of minor changes, including:
- batchprocessor -> concurrentbatchprocessor
- loggingexporter -> debugexporter
- eliminate arrow::disabled flags
- consistent port numbering
- working `printer` example (now with fewer dependencies)
- one fewer examples, there was a little redundancy.

Fixes #72.

### CODE REVIEW

**Comment by jmacd** [comment]
@codeboten deserves a bunch of credit for eliminating module inconsistencies in our dependencies! 🥳

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -50,34 +50,36 @@ export PATH="$PATH:$(go env GOPATH)/bin"
...
-
-While the use of replace statements works to enable local development,
-it prevents running code directly from the repository, which raises a
-barrier to entry.  To work around this problem, the checked-in
-contents of `./collector/cmd/otelarrowcol/go.mod` must not contain
-`replace` statements.  To build an `otelarrowcol` from locally
-modified sources requires uncommenting the `replaces` directive in
-`./collector/cmd/otelarrowcol/build-config.yaml` and re-running `make
-genotelarrowcol otelarrowcol`.
+This repository contains a top-level `go.work` file, as an experiment.
```
Comment (lquerel) [inline CONTRIBUTING.md]: Is it still an experiment?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,34 +50,36 @@ export PATH="$PATH:$(go env GOPATH)/bin"
...
+This enables the Go modules defined here to avoid relative replace
+statements, which interfere with the ability to run them via simple
+`go install` and `go run` commands.  The `go.work` file names the
+three module definitions inside this repository and allows them all to
+be used at once during local development.
+
+### Upgrading OpenTelemtry Collector dependencies
```
Comment (lquerel) [inline CONTRIBUTING.md]:
```suggestion
### Upgrading OpenTelemetry Collector dependencies
```

**Review by lquerel — APPROVED** [review]
Waou very nice PR!

**Comment by lquerel** [comment]
> @codeboten deserves a bunch of credit for eliminating module inconsistencies in our dependencies! 🥳
@jmacd @codeboten 

Thank you both. This PR is just what we needed to be able to open/launch this project to a wider audience!

File: CONTRIBUTING.md (hunk 1 above)
Comment (jmacd) [reply CONTRIBUTING.md]: Thanks. I think not, at this point. If we have troubles releasing v0.7 it'll be addressed in the next PR.

**Pull request merged**